crates/sui-source-validation-service/config.toml

```diff
@@ -1,5 +1,6 @@
 [[packages]]
 repository = "https://github.com/mystenlabs/sui"
+branch = "mainnet-v1.3.1"
```

Comment:
The system packages are a bit of a weird special case here, because they change in place.  Any thoughts on how to handle this in the long-term, short of restarting the service every time we do a protocol upgrade?  This is basically what we do today for the related problem of bumping the `framework/mainnet` and `framework/testnet` tracking branches, but it's quite fiddly.

In that case, I was thinking about eventually moving to a model where:

- We maintain a mapping of protocol versions to commits (we already have this, in fact).
- We use the `sui` CLI as an external resolver to convert the current protocol config version to a commit.

This is not directly applicable in this case, but just wanted share because it seems related.

Reply:
> short of restarting the service every time we do a protocol upgrade? 

yeah definitely don't want to have to kick the server every time. my thoughts so far are:

- trigger resolving by observable on-chain effect (protocol / package upgrade)

Then:

- use a tag naming and/or datetime convention. This sounds, in effect, similar to the mapping of protocol versions to commits, and the first thing to that comes to mind

- The part where `sui` CLI performs this is an interesting idea, and might be worth doing, but we can probably proceed without it for now. I'm also not sure how/if `sui` CLI would do this more generically for other Move packages?

Reply:
The CLI can't do this for other Move packages, but other packages also don't change on-chain once published -- that's the main thing that makes the system packages special.  There is a related problem though of the git repo for a package evolving and spinning off multiple packages (multiple publishes and/or upgrades on one chain or across many), that needs to be navigated.